tests/routes/test_sessions.py

```diff
@@ -591,14 +603,109 @@ def test_set_session_peers_with_limit(
+    )
+    assert response.status_code == 200  # Should succeed since no observers
+
+    # Test 2: Try to set 11 peers with observe_others=True (exceeds limit of 10)
+    peers_dict_all_observers: dict[str, dict[str, Any]] = {
+        peer_name: {"observe_others": True} for peer_name in peer_names[:11]
+    }
+    response = client.put(
+        f"/v2/workspaces/{test_workspace.name}/sessions/{session_id}/peers",
+        json=peers_dict_all_observers,
+    )
+    assert response.status_code == 422  # ValidationException
+    assert "11 observers" in response.json()["detail"]
+    assert "Maximum allowed is 10 observers" in response.json()["detail"]
+
+    # Test 3: Set exactly 10 observers should succeed
+    peers_dict_ten_observers: dict[str, dict[str, Any]] = {
+        peer_name: {"observe_others": True} for peer_name in peer_names[:10]
+    }
+    response = client.put(
+        f"/v2/workspaces/{test_workspace.name}/sessions/{session_id}/peers",
+        json=peers_dict_ten_observers,
+    )
+    assert response.status_code == 200  # Should succeed with exactly 10 observers
+
+
+def test_update_peer_config_observer_limit(
+    client: TestClient,
+    sample_data: tuple[Workspace, Peer],
+):
+    """Test that updating peer config respects observer limits"""
+    test_workspace, test_peer = sample_data
+
+    # Create a test session
+    session_id = str(generate_nanoid())
+    response = client.post(
+        f"/v2/workspaces/{test_workspace.name}/sessions",
+        json={
+            "id": session_id,
+        },
+    )
+    assert response.status_code == 200
+
+    # Create exactly 10 peers and add them as observers
+    peer_names: list[str] = []
+    for i in range(10):
+        peer_name = f"observer_{i}"
+        response = client.post(
+            f"/v2/workspaces/{test_workspace.name}/peers",
+            json={"name": peer_name, "metadata": {}},
+        )
+        assert response.status_code == 200
+        peer_names.append(peer_name)
+
+    # Add all 10 peers as observers
+    peers_dict_observers: dict[str, dict[str, Any]] = {
+        peer_name: {"observe_others": True} for peer_name in peer_names
+    }
+    # Also add the test_peer as non-observer
+    peers_dict_observers[test_peer.name] = {"observe_others": False}
+
     response = client.put(
         f"/v2/workspaces/{test_workspace.name}/sessions/{session_id}/peers",
-        json=peers_dict,
+        json=peers_dict_observers,
+    )
+    assert response.status_code == 200  # Should succeed with exactly 10 observers
+
+    # Now try to update test_peer to become an observer (would exceed limit)
+    response = client.post(
+        f"/v2/workspaces/{test_workspace.name}/sessions/{session_id}/peers/{test_peer.name}/config",
+        json={"observe_others": True},
+    )
+    assert response.status_code == 422  # ValidationException
+    assert "11 observers" in response.json()["detail"]
+    assert "Maximum allowed is 10 observers" in response.json()["detail"]
```

Comment:
I'm of the preference that tests not key into specific natural language elements of a response, and instead just confirm an HTTP status code -- but this is just an opinion